File: assignment1/chair.py
from typing import Sequence
from leg import Leg

class Chair:
    def __init__(self, numLegs : int, material : str, length : float = Leg.DEFAULT_LENGTH):
        self._legs : Sequence[Leg] = []
        if numLegs < 3:
            raise ValueError('Cannot create a chair with less than 3 legs.')
        for _ in range (0, numLegs):
            self._legs.append(Leg(length))
        
        self._numLegs = numLegs
        self._material : str = material

    @property    
    def material(self) -> str:
        return self._material

    @property
    def length(self) -> float:
        return self._legs[0].length

    @property
    def numLegs(self) -> int: 
        return self._numLegs

    @property    
    def broken(self):
        return self._legs[0].broken

    def sit(self, value : float):
        if self._legs[0].broken:
            raise ValueError("Chair is broken! Cannot sit.")
        if value > Leg.MAX_WEIGHT*self._numLegs:
            self._legs[0].broken = True

File: assignment2/src/chair.cpp
#include <stdexcept>
#include "chair.h"

Chair::Chair(int numLegs, std::string material, float length) {
    Chair::_numLegs = numLegs;
    Chair::_material = material;
    for(int i = 0; i < numLegs; i++)
        legs.push_back(Leg(length));
}

int Chair::getNumLegs() const { return _numLegs; }
float Chair::getLength() const { return Chair::legs[0].getLength(); } //come back tot his
std::string Chair::getMaterial() const { return _material; }
bool Chair::isBroken() const { return legs[0].isBroken(); }

void Chair::sit(float weight) {
    if (legs[0].isBroken())
        throw std::invalid_argument("Cannot sit, chair leg is broken.");
    if(weight > legs[0].MAX_WEIGHT*_numLegs)
        legs[0].setBroken(true);
}
